cmd/dockerd/daemon.go

```diff
@@ -135,19 +138,27 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 		return fmt.Errorf("Failed to load listeners: %v", err)
 	}
 
-	registryService, err := registry.NewService(cli.Config.ServiceOptions)
-	if err != nil {
-		return err
-	}
+	ctx, cancel := context.WithCancel(context.Background())
+	if cli.Config.ContainerdAddr == "" && runtime.GOOS != "windows" {
+		opts, err := cli.getContainerdDaemonOpts()
+		if err != nil {
+			cancel()
```

Comment:
It looks like if you move `defer cancel()` to right after `context.WithCancel` you won't have to call it explicitly, i.e.
```patch
	ctx, cancel := context.WithCancel(context.Background())
+ 	defer cancel()
￼ 	if cli.Config.ContainerdAddr == "" && runtime.GOOS != "windows" {
￼ 		opts, err := cli.getContainerdDaemonOpts()
￼ 		if err != nil {
- 			cancel()
￼ 			return fmt.Errorf("Failed to generate containerd options: %v", err)
￼ 		}
```

Reply:
This is actually like this on purpose, the defer ordering is important because the `cancel()` should run before `r.WaitTimeout(10 * time.Second)`. I should probably note this in a comment